tests/rptest/services/redpanda.py

```diff
@@ -977,6 +980,43 @@ def wait_for_membership(self, first_start):
                    backoff_sec=self._startup_poll_interval(first_start),
                    err_msg="Cluster membership did not stabilize")
 
+    def setup_azurite_dns(self):
+        """
+        Azure API relies on <container>.something DNS.  Doing DNS configuration
+        with docker/podman is unstable, as it isn't consistent across operating
+        systems.  Instead do it more crudely but robustly, but editing /etc/hosts.
+        """
+        azurite_ip = socket.gethostbyname("azurite")
+        azurite_dns = f"{self._si_settings.cloud_storage_azure_storage_account}.blob.localhost"
+
+        def update_hosts_file(node_name, path):
+            ducktape_hosts = open(path, "r").read()
+            if azurite_ip not in ducktape_hosts:
```

Comment:
Doesn't this have the same problem of matching other ip addresses that you highlighted on slack: "if the azurite container's IP was 1.2.3.4 then it would match existing entries like 1.2.3.46"?

Reply:
Yeah, I pushed a half-baked version, this is fixed now